pkg/proxy/handlers/spacelister_get_test.go

```diff
@@ -726,3 +742,234 @@ func TestGetUserWorkspace(t *testing.T) {
 		})
 	}
 }
+
+func TestSpaceListerGetPublicViewerEnabled(t *testing.T) {
+
+	fakeSignupService := fake.NewSignupService(
+		newSignup("batman", "batman.space", true),
+		newSignup("robin", "robin.space", true),
+	)
+
+	fakeClient := fake.InitClient(t,
+		// space
+		fake.NewSpace("robin", "member-1", "robin"),
+		fake.NewSpace("batman", "member-1", "batman"),
+
+		// spacebindings
+		fake.NewSpaceBinding("robin-1", "robin", "robin", "admin"),
+		fake.NewSpaceBinding("batman-1", "batman", "batman", "admin"),
+
+		fake.NewSpaceBinding("community-robin", toolchainv1alpha1.KubesawAuthenticatedUsername, "robin", "viewer"),
+	)
+
+	robinWS := workspaceFor(t, fakeClient, "robin", "admin", true)
+	batmanWS := workspaceFor(t, fakeClient, "batman", "admin", true)
+	publicRobinWS := func() *toolchainv1alpha1.Workspace {
+		batmansRobinWS := robinWS.DeepCopy()
+		batmansRobinWS.Status.Type = ""
+		batmansRobinWS.Status.Role = "viewer"
+		return batmansRobinWS
+	}()
+	tests := map[string]struct {
+		username          string
+		expectedErr       string
+		workspaceRequest  string
+		expectedWorkspace *toolchainv1alpha1.Workspace
+	}{
+		"robin can get robin workspace": {
+			username:          "robin.space",
+			workspaceRequest:  "robin",
+			expectedWorkspace: &robinWS,
+		},
+		"batman can get batman workspace": {
+			username:          "batman.space",
+			workspaceRequest:  "batman",
+			expectedWorkspace: &batmanWS,
+		},
+		"batman can get robin workspace as public-viewer": {
```

Comment:
What about adding a test when batman can't get the robin workspace as a public viewer because public viewer is not enabled in the context? You don't have this covered, right?

Reply:
done in a525db583e260bde5f677a42e6c1ebe3906c1e12